src/main/java/com/salesforce/storm/spout/dynamic/DynamicSpout.java

```diff
@@ -324,9 +306,6 @@ public void ack(Object id) {
 
         // Ack the tuple via the coordinator
         getCoordinator().ack(messageId);
-
-        // Update ack count metric for VirtualSpout this tuple originated from
```

Comment:
got moved to the virtual spout

Reply:
All of the metrics in this class that are scoped to `VirtualSpout` should go away. Most of them use the source spout id, I'd rather each `VirtualSpout` have it's own scoped metrics and than DynamicSpout have one that total stuff up.